### DISCUSSION TITLE
Linear Regression Message

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#270] [Q&A] [answered]
Permisi rekan-rekan semua.  

1. Saya mau bertanya penyebab nilai NA `Indian.Marble1` di `summary(priceHo_multi_model)` apa ya? 
![Screenshot (1754)](https://user-images.githubusercontent.com/83503249/184468076-ccf5a8c2-7460-48bc-b9a8-da1965acadb2.png)

2. Kemudian saya binggung kenapa nilai r.squared tepat 1 ?
3.  Lalu dibagian step wise dan prediction ada message berikut, kira-kira penyebabnya apa ya?  Kemudian bagaimana cara memperbaikinya ya?  

![Screenshot (1755)](https://user-images.githubusercontent.com/83503249/184468136-e5aefbfe-d4d6-4c52-aba9-66aac3d4c2d8.png)
![Screenshot (1757)](https://user-images.githubusercontent.com/83503249/184468138-bb997d40-bf8c-4553-b2fd-c7f631eac51c.png)

Oh iya sebelum step di atas saya sudah menghapus nilai-nilai duplikasi dan mengecek missing value. Saya juga sudah mencoba simple linear regression dengan 1 dan 2 prediktor yang korelasinya kuat tetapi r-squarednya masih rendah.
 
Sekian terima kasih banyak

**Comment by anugrah18003** [comment, marked as answer]
Dear @ToniAS21 ,

Untuk pertanyaan nomor 1, terkait penyebab nilai NA Indian.Marble1 di summary(priceHo_multi_model). Hal tersebut dapat terjadi karena adanya variabel lain yang bergantung linier (linearly dependent) dengan Indian.Marble1 sehingga Indian.Marble1 disini akan bersifat redundan dan hasil estimasi nya menjadi NA, silahkan untuk di cek kembali apakah terdapat varibel yang memiliki nilai korelasi yang sempurna, selain itu dapat di periksa juga untuk salah satu asumsi dari regresi linier yaitu asumsi non multikolinearitasnya dan pastikan tidak ada multikolinearitas.
Sebagai referensi agar lebih memahami, silahkan mengacu pada pembahasan berikut: https://stackoverflow.com/questions/49598084/how-do-i-fix-the-missing-output-na-in-my-summary-coefficients-table-r

Selanjutnya untuk pertanyaan nomor 2, terkait pengebab nilai r.squared tepat 1. Hal ini masih berkaitan dengan pertanyaan nomor 1, dimana nilai R-Square akan sangat besar ketika variabel yang digunakan saling bergantung linier (linearly dependent) dimana hal ini dapat ditandai dengan korelasi yang tinggi bahkan sempurna atau terdapat masalah multikolinearitas.

Selanjutnya pertanyaan nomor 3, warning pertama memberitahu bahwa  satu atau lebih variabel prediktor yang digunakan adalah prediktor respons yang hampir sempurna, sehingga tidak akan memiliki banyak ruang untuk memilih model yang berbeda. 
Sebagai referensi dapat mengacu pada diskusi berikut: https://www.mail-archive.com/search?l=r-help@r-project.org&q=subject:%22%5C%5BR%5C%5D+Error+in+a+regression%22&o=newest&f=1
Untuk warning kedua, bapak dapat mengikuti langkah-langkah berikut untuk mengatasi permasalahan yang terjadi:
https://www.statology.org/prediction-from-rank-deficient-fit-may-be-misleading/
atau
https://stackoverflow.com/questions/26558631/predict-lm-in-a-loop-warning-prediction-from-a-rank-deficient-fit-may-be-mis

Demikian yang dapat saya jelaskan, Semoga membantu Pak :)

**Reply by ToniAS21** [reply]
Baik terima kasih masukannya kak, tetapi saya masih binggung terkait bagaimana mengetahui kolom yang meyebabkan redundan ya? karena struktur data saya di dominasi jenis kategori, yang setau saya tidak bisa melihat korelasinya. Tadi saya juga sudah mencoba mengubah kolom `FirePlace` dan `Baths` menjadi numeric agar melihat korelasinya tetap hubunganya lemah.
![Screenshot (142)](https://user-images.githubusercontent.com/83503249/185190428-8a5e0647-4c06-44e4-aa05-c5b56592e32c.png)

Kemudian kolom `area` yang merupakan satu-satunya tipe numeric tidak memiliki korelasi yang kuat juga. 
![Screenshot (143)](https://user-images.githubusercontent.com/83503249/185191379-88736d8a-59ee-48c6-b9fb-37bc47bc8b7b.png)
 
Ini hasil asumsi non multikolinearitasnya kak.
![Screenshot (198)](https://user-images.githubusercontent.com/83503249/185194556-411b6078-cd0a-4078-8b1c-a99019aa5154.png)


Kemudian saya juga sudah mencoba menghapus kolom `Indian.Marble1` tetapi summary modelnya masih serupa juga seperti sebelumnya. 

Ini data yg saya gunakan kak, mungkin kalau mau dilihat.
https://www.kaggle.com/datasets/greenwing1985/housepricing/code

**Reply by anugrah18003** [reply]
Dear @ToniAS21 ,

Betul sekali pak ketika menggunakan fungsi dari cor() atau ggcorr() di R/RStudio hanya bisa digunakan oleh data bertipe numeric.

Berikut adalah hasil korelasi ketika datanya belum diubah tipe datanya, masih dalam bentuk numeric (double/int):

![image](https://user-images.githubusercontent.com/70314445/185307433-229c7f7f-e9f1-4b49-9f36-f5bf13103f55.png)

Sebagai gambaran awal bahwa setiap variabel hanya berkorelasi dengan Price (Price adalah target variabel sehingga bagus ketika ada korelasi variabel independen terhadap Price), kecuali pada White.Marble, Black.Marble dan Indian.Marble yang masing-masing berkorelasi sebesar -0.5. Tentu ini belum valid karena tipe data seharusnya dari ketiga variabel yang memiliki korelasi tersebut bertipe faktor/kategorik. Oleh karena itu akan dilakukan pengujian korelasi hanya pada ketiga variabel tersebut.

Salah satu metoe yang dapat digunakan untuk menguji korelasi antara variabel yang bertipe kateorik adalah Uji Pearson's Chi-squared test.

Dimana hipotesisnya sebagai berikut :

- H0 : tidak ada hubungan antara kedua variabel.
- H1 : terdapat hubungan antara kedua variabel.

![image](https://user-images.githubusercontent.com/70314445/185307527-22986ae3-9df7-4d54-9602-4feca1dbd161.png)

Dapat diketahui bahwa masing-masing dari ketiga variabel memiliki korelasi yang signifikan.

Mungkin sarannnya adalah untuk mempertimbangkan kembali apakah akan menggunakan satu atau dua saja dari variabel-varaibel marble tersebut.

**Reply by shinddayo** [reply]
Mas @anugrah18003 dari  hasil Uji Pearson's Chi-squared test nya mana kah yang menunjukkan korelasinya signifikan? x-squared nya? p-value nya?

**Reply by anugrah18003** [reply]
Dear @shinddayo , betul sekali untuk kriteria uji yang dapat digunakan dalam mengambil kesimpulan adalah Chi-squared (X-squared) atau p-value, untuk pengujian menggunakan X-squared bisa dibandingkan dengan nilai X-squared dari tabel X-squared dimana apabila nilai X-squared yang didapat dari perhitungan R lebih besar dibanding X-squared dari tabel maka H0 akan ditolak, untuk pengujian menggunakan p-value bisa dibandingkan dengan taraf signifikansi (alpha = 0.05), dimana apabila nilai p-value yang didapat dari perhitungan R lebih kecil dibanding alpha maka H0 akan ditolak.
Note: Agar lebih mudah cukup melihat salah satu kriteria uji saja, karena kesimpulan yang dihasilkan akan sama. Dan agar memudahkan lebih baik menggunakan nilai p-value karena pada umumnya taraf signifikan dalam penelitian adalah alpha = 0.05 sehingga tidak perlu menghitung kembali.

**Reply by ToniAS21** [reply]
Baik kak terima kasih banyak modelnya sudah oke
